### PR TITLE
Bitfit

### PR DESCRIPTION
**Pull request by lintangsutawika**

### CODE REVIEW

**Comment by lintangsutawika** [comment]
In `madx_run_clm.py` I added a new arg `finetuning_strategies` that can be extended to choose any other parameter-efficient finetuning methods.

**Comment by yongzx** [comment]
I will review this! Thanks!

> In madx_run_clm.py I added a new arg finetuning_strategies that can be extended to choose any other parameter-efficient finetuning methods.

It's already in there: `lang_adapt_strategies`.

File: scripts/lang_adapt/bitfit/train_tokenizer_update.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#SBATCH --mem=50g
+
+
+
+lng=$1
+model=$2
+tokenizer_dir=$3
+vocab_size=$4
+sample_size=$5
+python tokenized4clm_sampled.py --lang $lng --model $model --tokenizer_dir $tokenizer_dir --vocab_size $vocab_size --sample_size $sample_size --extend_vocab
```
Comment (yongzx) [inline scripts/lang_adapt/bitfit/train_tokenizer_update.sh:16]: Is it just the batch script for training tokenizer?

**Review by yongzx — CHANGES_REQUESTED** [review]

File: scripts/lang_adapt/madx_run_clm.py [hunk 2]
```diff
@@ -111,6 +111,10 @@ class ModelArguments:
         default="",
         metadata={"help": "choose one of the two strategies - 'replace', 'extend', 'overlap-replace'"},
     )
+    finetuning_strategies: str = field(
```
[the remark this answers is not in the document]
Comment (yongzx) [reply scripts/lang_adapt/madx_run_clm.py:114]: Also, do add "bitfit" to `assert model_args.lang_adapt_strategies in ('emb', 'emb-and-adpt', 'emb-then-adpt', 'lora')` (in the latest master branch)

**Comment by yongzx** [comment]
@lintangsutawika Left some reviews. Everything else LGTM.

**Comment by lintangsutawika** [comment]
@yongzx Bitfit isn't really an adapter, it just freezes the biases of a given model. I'm not sure it fits in the `modify_model`.

I've placed it it line 625 currently. 
```
        elif model_args.lang_adapt_strategies == "bitfit":
            for name, param in model.transformer.named_parameters():
                if 'bias' not in name:
                    param.requires_grad = False
```

**Comment by yongzx** [comment]
Thanks Lintang! I will run it today and merge it by EOD.

**Pull request merged**